File: stage-02th/src/main/java/cn/uncleyumo/chapter06/unit05/Test02.java
package cn.uncleyumo.chapter06.unit05;

import cn.uncleyumo.utils.ColorPrinter;
import cn.uncleyumo.utils.LogPrinter;

import java.io.*;

/**
 * @author dev.uncleyumo.cn(BookAir13)
 * @fileName Test02
 * @createDate 4/28/25 April
 * @school Wuxi University
 * @studentID 22344131
 * @description
 */

public class Test02 {
    public static String staticPath = Test01.staticPath;
    public static void main(String[] args) {
        practise02();
    }

    private static void practise02() {
        try (
                Writer writer = new FileWriter(staticPath + "a.txt");
                BufferedWriter bw = new BufferedWriter(writer)
        ) {
            bw.write("Huawei好呀Huawei美".toCharArray());
            bw.newLine();
            bw.write("华为给我增智慧".toCharArray());
            bw.flush();
        } catch (Exception e) {
            LogPrinter.INSTANCE.error(e.getMessage());
        }
    }

    private static void practise01() {
        try (
                Reader fileReader = new FileReader(staticPath + "a.txt");
                BufferedReader br = new BufferedReader(fileReader)
        ) {
            System.out.println(br.readLine());
            System.out.println(br.readLine());
            System.out.println("---------------------------------------------");
            char[] buffer = new char[3];
            int len;
            while ((len = br.read(buffer)) != -1) {
                ColorPrinter.INSTANCE.printFontCyan(new String(buffer, 0, len));
            }
        } catch (Exception e) {
            LogPrinter.INSTANCE.error(e.getMessage());
        }
    }
}

File: stage-02th/src/main/java/cn/uncleyumo/chapter09/unit01/kotlin/Test02.kt
@file:Suppress("NonAsciiCharacters", "FunctionName", "ClassName", "LocalVariableName", "PropertyName")

package cn.uncleyumo.chapter09.unit01.kotlin

/**
 * @author dev.uncleyumo.cn(BookAir13)
 * @fileName Test02
 * @createDate 5/12/25 May
 * @school Wuxi University
 * @studentID 22344131
 * @description
 */

interface 动物接口 {
    val 姓名: String
    val 年龄: Int
    fun 吃东西()
    fun 睡觉()
    fun 获取显示信息() = "姓名：${姓名}，年龄：${年龄}"
}

class 狗(override val 姓名: String, override val 年龄: Int) : 动物接口 {
    override fun 吃东西() = 打印("狗吃狗粮")
    override fun 睡觉() = 打印("狗睡觉")
}

class 猫(override val 姓名: String, override val 年龄: Int) : 动物接口 {
    override fun 吃东西() = 打印("猫吃猫粮")
    override fun 睡觉() = 打印("猫睡觉")
}

fun 打印(文字: String) = println(文字)

fun 主函数() {
    val 动物: 动物接口 = 狗("旺财", 3)
//    val 动物: 动物接口 = 猫("小花", 2)
    动物.吃东西()
    动物.睡觉()
    打印(动物.获取显示信息())
}

fun main() {
    打印("主函数开始")
    主函数()
    打印("主函数结束")
}